modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -1308,22 +1314,39 @@ private void _bindObjectDefinitions(ObjectRelationship objectRelationship)
 			_objectDefinitionPersistence.findByPrimaryKey(
 				objectRelationship.getObjectDefinitionId2());
 
-		String objectDefinition2PreviousRESTContextPath =
-			objectDefinition2.getRESTContextPath();
+		Tree tree = _treeFactory.createObjectDefinitionTree(
```

Comment:
Before starting the logic involving objectDefinition2, can we move this code to here:
```
if (objectDefinition1.isApproved()) {
	objectDefinition1.setPreviousRESTContextPath(
		objectDefinition1PreviousRESTContextPath);
	objectDefinitionLocalService.deployObjectDefinition(
		objectDefinition1);
}
```
This method would have a block of logic about objectDefinition1 and after objectDefinition2.
What do you think? can we test it?

Reply:
Also, related to this condition, I noticed that @carolmariaabb did this [change](https://github.com/liferay-objects/liferay-portal/pull/3667/commits/a6e9a0b475a636182c39e60a7c2d596d0ed36705) in her PR. Can we please check if it is necessary so you can add it to your PR?